pkg/controllers/controlplanemachineset/cluster_operator_test.go

```diff
@@ -128,29 +148,211 @@ var _ = Describe("Cluster Operator Status with a running controller", func() {
 				})).Should(Succeed())
 			})
 
-			PIt("Set the cluster operator available", func() {
-				co := resourcebuilder.ClusterOperator().WithName(operatorName).Build()
-
-				Eventually(komega.Object(co)).Should(HaveField(".Status.Conditions", test.MatchClusterOperatorStatusConditions([]configv1.ClusterOperatorStatusCondition{
+			It("Set the cluster operator available", func() {
+				Eventually(komega.Object(co)).Should(HaveField("Status.Conditions", test.MatchClusterOperatorStatusConditions([]configv1.ClusterOperatorStatusCondition{
 					{
-						Type:   configv1.OperatorAvailable,
-						Status: configv1.ConditionTrue,
+						Type:    configv1.OperatorAvailable,
+						Status:  configv1.ConditionTrue,
+						Reason:  reasonAsExpected,
+						Message: "cluster operator is available",
 					},
 					{
 						Type:   configv1.OperatorProgressing,
 						Status: configv1.ConditionFalse,
+						Reason: reasonAsExpected,
 					},
 					{
 						Type:   configv1.OperatorDegraded,
 						Status: configv1.ConditionFalse,
+						Reason: reasonAsExpected,
 					},
 					{
-						Type:   configv1.OperatorUpgradeable,
-						Status: configv1.ConditionTrue,
+						Type:    configv1.OperatorUpgradeable,
+						Status:  configv1.ConditionTrue,
+						Reason:  reasonAsExpected,
+						Message: "cluster operator is upgradable",
 					},
 				})))
 			})
 		})
+	})
+})
+
+var _ = Describe("Cluster Operator Status", func() {
+	const operatorName = "control-plane-machine-set"
+	var co *configv1.ClusterOperator
+	var reconciler *ControlPlaneMachineSetReconciler
+	var logger test.TestLogger
+	var namespaceName string
+
+	cpmsBuilder := resourcebuilder.ControlPlaneMachineSet().WithName(clusterControlPlaneMachineSetName).WithNamespace(namespaceName)
+
+	BeforeEach(func() {
+		By("Setting up a namespace for the test")
+		ns := resourcebuilder.Namespace().WithGenerateName("control-plane-machine-set-cluster-operator-").Build()
+		Expect(k8sClient.Create(ctx, ns)).To(Succeed())
+		namespaceName = ns.GetName()
+
+		reconciler = &ControlPlaneMachineSetReconciler{
+			Client:       k8sClient,
+			Namespace:    namespaceName,
+			OperatorName: operatorName,
+		}
+
+		// CVO will create a blank cluster operator for us before the operator starts.
+		co = resourcebuilder.ClusterOperator().WithName(operatorName).Build()
+		Expect(k8sClient.Create(ctx, co)).To(Succeed())
 
+		logger = test.NewTestLogger()
+	})
+
+	AfterEach(func() {
+		test.CleanupResources(Default, ctx, cfg, k8sClient, namespaceName,
+			&configv1.ClusterOperator{},
```

Comment:
You also need to delete the CPMSs
```suggestion
			&configv1.ClusterOperator{},
			&machinev1.ControlPlaneMachineSet{},
```